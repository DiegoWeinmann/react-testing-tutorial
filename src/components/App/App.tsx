import React from 'react';
import Counter from '../Counter/Counter';

function App() {
  return (
    <div>
      <h1>React App</h1>
      <Counter />
    </div>
  );
}

export default App;
