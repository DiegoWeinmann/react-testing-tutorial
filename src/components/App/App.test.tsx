import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';

afterEach(cleanup);

describe('<App/>', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
