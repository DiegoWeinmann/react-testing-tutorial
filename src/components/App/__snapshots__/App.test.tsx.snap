// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      React App
    </h1>
    <h1
      data-testid="counter"
    >
      0
    </h1>
    <button
      data-testid="button-up"
    >
      Up
    </button>
    <button
      data-testid="button-down"
      disabled=""
    >
      Down
    </button>
  </div>
</DocumentFragment>
`;
